diff --git a/lrexlib-pcre2-2.9.1-1.rockspec b/lrexlib-pcre2-2.9.1-1.rockspec
index ea788041..9ca5fbf9 100644
--- a/lrexlib-pcre2-2.9.1-1.rockspec
+++ b/lrexlib-pcre2-2.9.1-1.rockspec
@@ -13,12 +13,14 @@ build = {
       },
       libraries = {
         "pcre2-8",
+        "luajit-5.1",
       },
       incdirs = {
         "$(PCRE2_INCDIR)",
       },
       libdirs = {
         "$(PCRE2_LIBDIR)",
+        "$(STAGE_EMBEDDED_LIBDIR)",
       },
     },
   },
