diff --git a/configure b/configure
index 0a4f09b..e7d0d0c 100755
--- a/configure
+++ b/configure
@@ -1164,38 +1164,6 @@ _EOC_
         }
         push @make_install_cmds, "cd $root_dir/build/$resty_cli_dir && "
              . "\$(INSTALL) bin/* $target_dir";
-
-        if ($platform ne 'msys') {
-            # patch the resty script:
-
-            print "patching the resty script with hard-coded nginx binary ",
-                  "path...\n";
-
-            my $resty_bin = "$root_dir/build/$resty_cli_dir/bin/resty";
-            open my $in, $resty_bin
-                or die "Cannot open $resty_bin for reading: $!\n";
-            my ($new, $found);
-            while (<$in>) {
-                if (/^my \$nginx_path;$/) {
-                    $found = 1;
-                    $new .= qq/my \$nginx_path = '$ngx_sbin';\n/;
-
-                } else {
-                    $new .= $_;
-                }
-            }
-            close $in;
-
-            if (!$found) {
-                die "failed to patch $resty_bin: the line 'my \$nginx_path' ",
-                    "was not found.\n";
-            }
-
-            open my $out, ">$resty_bin"
-                or die "Cannot open $resty_bin for writing: $!\n";
-            print $out $new;
-            close $out;
-        }
     }
 
     # configure restydoc indexes
