diff --git a/bundle/opm-0.0.6/bin/opm b/bundle/opm-0.0.6/bin/opm
index 9029225..24a8200 100755
--- a/bundle/opm-0.0.6/bin/opm
+++ b/bundle/opm-0.0.6/bin/opm
@@ -97,10 +97,6 @@ if ($cmd eq '-v') {
     exit;
 }
 
-# explicitly clear the environments to avoid breaking luajit and resty.
-delete $ENV{LUA_PATH};
-delete $ENV{LUA_CPATH};
-
 for ($cmd) {
     if ($_ eq 'get' || $_ eq 'install') {
         check_lock_file();
