diff --git a/proxy/http/HttpTransact.cc b/proxy/http/HttpTransact.cc
index 813dfdae1..f849084c3 100644
--- a/proxy/http/HttpTransact.cc
+++ b/proxy/http/HttpTransact.cc
@@ -6477,7 +6477,7 @@ HttpTransact::is_request_retryable(State *s)
   // If safe requests are  retryable, it should be safe to retry safe requests irrespective of bytes sent or connection state
   // according to RFC the following methods are safe (https://tools.ietf.org/html/rfc7231#section-4.2.1)
   // Otherwise, if there was no error establishing the connection (and we sent bytes)-- we cannot retry
-  if (!HttpTransactHeaders::is_method_safe(s->method) && s->current.state != CONNECTION_ERROR &&
+  if (s->current.state != CONNECTION_ERROR &&
       s->state_machine->server_request_hdr_bytes > 0) {
     return false;
   }
